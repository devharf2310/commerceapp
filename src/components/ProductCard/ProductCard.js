import React from 'react';
import PropTypes from 'prop-types';
import { Card, CardTitle, CardText, Grid, Cell } from "react-md";
import './ProductCard.scss';


const ProductCard = ({product}) => (

    <Card className="md-block-centered">
        <CardTitle
            title={product.name}
            subtitle={`${product.categories.join(", ")} - ${product.brand}`}
        />
        <CardText>
        <Grid>
            <Cell size={4}>
            <center>
                <img alt={product.name} src={product.photo} className="product-image" />
            </center>
            </Cell>
            <Cell size={8}>
                <p>{product.description}</p>
                <p><span className="span-product-detail">Stock:</span> {product.stock}</p>
                <p><span className="span-product-detail">Price:</span> <span className="currency-value">{product.price}</span></p>
            </Cell>
        </Grid>
        </CardText>
    </Card>
);


ProductCard.propTypes = {
    product : PropTypes.shape({name: PropTypes.string.isRequired,
    categories: PropTypes.array.isRequired,
    brand: PropTypes.string.isRequired,
    photo: PropTypes.string.isRequired,
    stock: PropTypes.number.isRequired,
    price: PropTypes.string.isRequired
  })};

export default ProductCard;
