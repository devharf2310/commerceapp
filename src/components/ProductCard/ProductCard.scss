.span-product-detail{
    font-weight: bolder;
}
