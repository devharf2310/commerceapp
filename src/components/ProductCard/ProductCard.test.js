import React from 'react'
import renderer from 'react-test-renderer'
import ProductCard from './ProductCard';

const product = {
    "id": "c9a5c012-6295-48d5-a103-68ca76ff047b",
    "name": "Intelligent Concrete Computer",
    "description": "Omnis veritatis in aliquam aspernatur nulla. Officia quidem qui. Maxime veniam nulla sint qui dolorem qui saepe neque placeat. Magni culpa aut eos molestiae nihil molestias. Quidem consectetur quia quo blanditiis perspiciatis facere dolorem.",
    "price": "752.00",
    "brand": "Gusikowski - Ebert",
    "stock": 44903,
    "photo": "http://lorempixel.com/640/480/technics",
    "categories": ["Services"]
  };

describe('product card snapshot tests with data', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ProductCard product={product}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

