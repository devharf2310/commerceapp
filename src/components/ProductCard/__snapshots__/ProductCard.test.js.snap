// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product card snapshot tests with data renders correctly 1`] = `
<div
  className="md-paper md-paper--1 md-card md-background--card md-block-centered"
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="md-card-title md-card-title--primary"
  >
    <div
      className="md-card-title--title-block"
    >
      <h2
        className="md-card-title--title md-card-title--large md-text"
        tabIndex={null}
      >
        Intelligent Concrete Computer
      </h2>
      <h3
        className="md-card-title--title md-text--secondary"
      >
        Services - Gusikowski - Ebert
      </h3>
    </div>
  </div>
  <section
    className="md-card-text"
  >
    <div
      className="md-grid"
    >
      <div
        className="md-cell md-cell--4"
      >
        <center>
          <img
            alt="Intelligent Concrete Computer"
            className="product-image"
            src="http://lorempixel.com/640/480/technics"
          />
        </center>
      </div>
      <div
        className="md-cell md-cell--8"
      >
        <p>
          Omnis veritatis in aliquam aspernatur nulla. Officia quidem qui. Maxime veniam nulla sint qui dolorem qui saepe neque placeat. Magni culpa aut eos molestiae nihil molestias. Quidem consectetur quia quo blanditiis perspiciatis facere dolorem.
        </p>
        <p>
          <span
            className="span-product-detail"
          >
            Stock:
          </span>
           
          44903
        </p>
        <p>
          <span
            className="span-product-detail"
          >
            Price:
          </span>
           
          <span
            className="currency-value"
          >
            752.00
          </span>
        </p>
      </div>
    </div>
  </section>
</div>
`;
